public/contracts/BountyDistribution.sol

```diff
@@ -1,37 +1,183 @@
+        require(membership[msg.sender]); 
+        _;
+    }
+
+    // Congress Variables/Structs
+    mapping (address => bool) membership; 
+    mapping (address => MemberProposal) addProposals; 
+    mapping (address => MemberProposal) removeProposals; 
+
+    uint numMembers; 
+    uint debatingPeriodInMinutes = 600; 
+
+    struct MemberProposal { 
+        address recipient; 
+        uint minExecutionDate; 
+        uint numberOfVotes; 
+        int currentResult; 
+        bool open; 
+        bool isValid;
+        
+        mapping (address => bool) voted; 
+    }
+
+    // Q&A Variables/Structs
     struct Question {
         uint256 hash;
         uint256 bounty; // denominated in WEI. 1 ether = 10^18 wei
         address askerAddr;
-        uint256 answerHash;
+        mapping(uint256 => Answer) answers;
+        mapping(uint256 => uint256) indexes; 
+        uint minExecutionDate; 
+
+        uint numAnswers; 
         bool settled;
         bool isValue; // checks for existence of key
     }
 
+    struct Answer {
+        address answerAddr; 
+        uint256 answerHash; 
+        
+        uint256 numUpvotes; 
+        bool isValue;
+        mapping(address => bool) voted; 
+    }
+
     mapping (uint256 => Question) public questions;
-    address contractCreator;
 
-    function BountyDistribution() public { // constructor
-        contractCreator = msg.sender;
+    constructor() public {
+        membership[msg.sender] = true; 
+    }
+
+    // Membership Functions
+    function newProposal(address recipient, bool addAddress) onlyMembers public {
+        if (addAddress) {
+            require(!addProposals[recipient].isValid || !addProposals[recipient].open);
+        } else {
+            require(!removeProposals[recipient].isValid || !removeProposals[recipient].open);
+        }
+
+        if (addAddress) { 
+            addProposals[recipient] = MemberProposal({
+                recipient: recipient,
+                minExecutionDate: now + debatingPeriodInMinutes * 1 minutes,
+                numberOfVotes: 0,
+                currentResult: 0,
+                open: true,
+                isValid: true
+            });
+        } else {
+            removeProposals[recipient] = MemberProposal({
+                recipient: recipient,
+                minExecutionDate: now + debatingPeriodInMinutes * 1 minutes,
+                numberOfVotes: 0,
+                currentResult: 0,
+                open: true,
+                isValid: true
+            });
+        }
     }
 
-    function distributeBounty(address _answerer, uint256 _aHash, uint256 _qHash) public {
+    function vote(address recipient, bool add, bool support) onlyMembers public {
+        MemberProposal storage p; 
+        if (add) { 
+            p = addProposals[recipient];
+        } else {
+            p = removeProposals[recipient]; 
+        }
+        require (p.isValid); 
+
+        require(!p.voted[msg.sender]); 
+        p.numberOfVotes++; 
```

Comment:
needs to add voter address to the `voted` mapping

Reply:
resolved.